Sprint-3/alarmclock/alarmclock.js

```diff
@@ -1,4 +1,42 @@
-function setAlarm() {}
+function setAlarm() {
+  const input = document.getElementById("alarmSet").value;
+  let timeRemaining = parseInt(input, 10);
+
+  if (isNaN(timeRemaining) || timeRemaining < 0) {
+    alert("Please enter a valid number of seconds.");
+    return;
+  }
+
+  // Update the display immediately
+  updateDisplay(timeRemaining);
+
+  // Clear any existing interval
+  if (window.alarmInterval) clearInterval(window.alarmInterval);
+
+  // Start countdown
+  window.alarmInterval = setInterval(() => {
+    timeRemaining--;
+
+    updateDisplay(timeRemaining);
+
+    if (timeRemaining <= 0) {
```

Comment:
After the alarm goes off and I hit 'Stop Alarm', I expected the input to be cleared too. This might be a nice user experience (UX) change to implement. What do you think?